sqfentity/lib/sqfentity.dart

```diff
@@ -375,12 +385,27 @@ class SqfEntityProvider extends SqfEntityModelBase {
       if (openedBatch[_dbModel!.databaseName!] == null || ignoreBatch) {
         final Database db = (await this.db)!;
         result = await db.rawInsert(pSql, params);
+        if (_dbModel!.postSaveAction != null) {
+          final openBracket = pSql.split('(')[1];
+          final closeBracket = openBracket.split(')')[0];
+
+          final keys = closeBracket.split(',');
+
+          final record = {};
+
+          for (int i = 0; i < params!.length; i++) {
+            record[keys[i]] = params[i];
+          }
+
+          _dbModel!.postSaveAction!(_tableName!, record, 'INSERT');

```

Comment:
we need to wait for `postSaveAction` to complete before ending the save. 
```suggestion
         await _dbModel!.postSaveAction!(_tableName!, record, 'INSERT');
```

Reply:
This `rawInsert` method is also called on update. The `pSql` string can also contain `OR REPLACE` if it's an update query. 

We can use this to determine if the post action is an `INSERT` OR `UPDATE`.  Lets use `INSERT` if its an insert and `UPSERT` if its an update

```suggestion
          await _dbModel!.postSaveAction!(_tableName!, record, pSql.include('OR REPLACE') ? 'UPSERT' : 'INSERT');
```